src/pyodide/internal/_workers.py

```diff
@@ -930,6 +931,47 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class _WorkflowStepWrapper:
+    def __init__(self, js_step):
+        self._js_step = js_step
+
+    def do(self, name, callback=None, config=None):
+        if callback is None:
+            # Return a decorator that will execute the function when called
+            def decorator(func):
+                @functools.wraps(func)
+                async def wrapper(*args, **kwargs):
+                    return self._do_call(name, config, func)
```

Comment:
Let's move `_do_call` to a top level function instead of a method and call it like:
```suggestion
                    return _do_call(self, name, config, func)
```